exercises/alphametics/example.rs

```diff
@@ -0,0 +1,70 @@
+// This is a brute-force solution, use `cargo test --release` for faster testing
+
+extern crate itertools;
+extern crate permutohedron;
+
+use itertools::Itertools;
+use permutohedron::Heap as Permutations;
+
+
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::char;
+
+fn test_equation(puzzle: &str, substitutions: &HashMap<char, u8>) -> bool {
+    // Create a new String with characters changed to numbers
+    let puzzle: String = puzzle.chars().map(|c| {
+        if let Some(&n) = substitutions.get(&c) {
+            // If the character is in the substitutions, get the number and
+            // convert it to a char
+            char::from_digit(n as u32,10).unwrap()
+        } else {
+            // Otherwise just copy over the character
+            c
```

Comment:
The else branch will run every time an equation is parsed. This will convert a String like `"A + B == C"` to another String which looks like this `"1 + 2 == 3"`. The the function chops up this string later and parses the individual numbers.